addons/api/addon/adapters/application.js

```diff
@@ -124,9 +124,16 @@ export default class ApplicationAdapter extends RESTAdapter.extend(
       result = await super.query(store, schema, query);
       //add the result items to a data array
       if (result && result.items) {
+        //filter out the removed_ids before returning the result set
+        //This is due to items being removed from the previous iteration
+        //while the user is still paginating
+        if (result.removed_ids) {
+          data = data.items.filter((i) => !result.removed_ids.includes(i));
+        }
         data.push(...result.items);
         //pass in the refresh token for subsequent calls to fetch the remaining list items
         query.refresh_token = result.refresh_token;
+
         //break the loop as soon the response_type becomes complete
         if (result.response_type === 'complete') {
```

Comment:
Not related to this PR but why is this `break` needed? Isn't it already handled by your while loop?

Reply:
addressed [here](https://github.com/hashicorp/boundary-ui/pull/2032)